### ISSUE TITLE
kohya-LoRA-trainer-XL.ipynb Lora degrades in the process of training, but does not learn

### ISSUE THREAD

**Issue by drimeF0** [Linaqruf/kohya-trainer#291]
I use the repository from qaneel

![изображение](https://github.com/Linaqruf/kohya-trainer/assets/67423319/36b6c1f1-7580-4bb3-8009-e16df0e0622c)

![изображение](https://github.com/Linaqruf/kohya-trainer/assets/67423319/6746b363-7c92-43ef-acd7-64df0bb154c1)

![изображение](https://github.com/Linaqruf/kohya-trainer/assets/67423319/772cb033-543f-4ed4-9d24-0e0a55352dee)

config:

```
[sdxl_arguments]
cache_text_encoder_outputs = false
no_half_vae = true
min_timestep = 0
max_timestep = 1000
shuffle_caption = true
lowram = true

[model_arguments]
pretrained_model_name_or_path = "gsdf/CounterfeitXL"
vae = "/content/vae/sdxl_vae.safetensors"

[dataset_arguments]
debug_dataset = false
in_json = "/content/LoRA/meta_lat.json"
train_data_dir = "/content/LoRA/train_data"
dataset_repeats = 1
keep_tokens = 0
resolution = "1024,1024"
color_aug = false
token_warmup_min = 1
token_warmup_step = 0

[training_arguments]
output_dir = "/content/LoRA/output/sdxl_lora"
output_name = "sdxl_lora"
save_precision = "fp16"
save_every_n_epochs = 1
train_batch_size = 4
max_token_length = 225
mem_eff_attn = false
sdpa = true
xformers = false
max_train_epochs = 10
max_data_loader_n_workers = 8
persistent_data_loader_workers = true
gradient_checkpointing = true
gradient_accumulation_steps = 1
mixed_precision = "fp16"

[logging_arguments]
log_with = "tensorboard"
logging_dir = "/content/LoRA/logs"
log_prefix = "sdxl_lora"

[sample_prompt_arguments]
sample_every_n_epochs = 1
sample_sampler = "euler_a"

[saving_arguments]
save_model_as = "safetensors"

[optimizer_arguments]
optimizer_type = "AdaFactor"
learning_rate = 0.0001
max_grad_norm = 0
optimizer_args = [ "scale_parameter=False", "relative_step=False", "warmup_init=False",]
lr_scheduler = "constant_with_warmup"
lr_warmup_steps = 100

[additional_network_arguments]
no_metadata = false
network_module = "networks.lora"
network_dim = 32
network_alpha = 16
network_args = []
network_train_unet_only = true

```

The problem is that after each epoch, the quality of generation breaks down dramatically, and at about 3-4 epochs, a terrible mess begins.

Here is an example from the first training epoch:
![Без имени](https://github.com/Linaqruf/kohya-trainer/assets/67423319/76b34aac-60fa-4bba-a96f-2a470d6e1e98)

And here is an example already at the 2nd epoch of training:
![Без имени](https://github.com/Linaqruf/kohya-trainer/assets/67423319/cb3f2060-1a45-4532-baed-71453aae1072)

And finally, the 8th epoch:
![Без имени](https://github.com/Linaqruf/kohya-trainer/assets/67423319/a28ced33-e15d-4d7e-a32a-b93a70461fe7)

What am I doing wrong, maybe I need to increase the number of epochs?

**Comment by drimeF0** [comment]
Today I checked the main code of kohya_ss, the results are the same, so I'm closing the issue here

**Comment by Linaqruf** [comment]
The problem is probably the datasets, also don't rely on training sample.

**Comment by drimeF0** [comment]
> The problem is probably the datasets, also don't rely on training sample.

I've tried it 10 times already with different datasets and it always ends up with a terrible result. Today I’ll try to manually select images for training and see what happens

**Comment by Linaqruf** [comment]
or use lower learning rate

**Comment by drimeF0** [comment]
> The problem is probably the datasets, also don't rely on training sample.
![Без имени](https://github.com/Linaqruf/kohya-trainer/assets/67423319/b54103e7-0e6a-4348-87eb-0aca63d0f126)
100 learning steps, 48 ​​pictures from safebooru with the tag "one_eye_closed", I already used my notebook, and apparently I made everything even worse than before.
I also noticed this:
![firefox_IzvCz2GJgV](https://github.com/Linaqruf/kohya-trainer/assets/67423319/e2fb42d5-019b-48a9-8b28-4eeda4a8978e)
Could this be a problem with such terrible image quality?

**Comment by drimeF0** [comment]
> or use lower learning rate

by 0.00002, the deterioration of the quality of the generated images seems to be much slower.  I'll see what happens after 1000 steps of learning.

**Comment by drimeF0** [comment]
> > or use lower learning rate
> 
> by 0.00002, the deterioration of the quality of the generated images seems to be much slower. I'll see what happens after 1000 steps of learning.

the image style has changed, but LoRA has not yet learned the concept of the one_eye_closed tag
![Без имени](https://github.com/Linaqruf/kohya-trainer/assets/67423319/109b12e6-fac2-4942-b450-de136f78bcd5)

**Comment by Linaqruf** [comment]
> > The problem is probably the datasets, also don't rely on training sample.
> > ![Без имени](https://user-images.githubusercontent.com/67423319/267960872-b54103e7-0e6a-4348-87eb-0aca63d0f126.png)
> > 100 learning steps, 48 ​​pictures from safebooru with the tag "one_eye_closed", I already used my notebook, and apparently I made everything even worse than before.
> > I also noticed this:
> > ![firefox_IzvCz2GJgV](https://user-images.githubusercontent.com/67423319/267961939-e2fb42d5-019b-48a9-8b28-4eeda4a8978e.png)
> > Could this be a problem with such terrible image quality?

yes, 512x512 is unusable for sdxl

**Comment by drimeF0** [comment]
![Без имени](https://github.com/Linaqruf/kohya-trainer/assets/67423319/4a8840de-6fd4-488f-8fcc-3e8261dfc10c)


> or use lower learning rate

I changed the regular lora to lora_fa from https://github.com/bmaltais/kohya_ss and it worked immediately after just 50 learning steps
Also, I set lr to 0.00005

**Comment by Linaqruf** [comment]
0.0002 or 2e-4 is higher than 1e-4, maybe 1e-5 or 5e-5

**Comment by drimeF0** [comment]
> > > The problem is probably the datasets, also don't rely on training sample.
> > > ![Без имени](https://user-images.githubusercontent.com/67423319/267960872-b54103e7-0e6a-4348-87eb-0aca63d0f126.png)
> > > 100 learning steps, 48 ​​pictures from safebooru with the tag "one_eye_closed", I already used my notebook, and apparently I made everything even worse than before.
> > > I also noticed this:
> > > ![firefox_IzvCz2GJgV](https://user-images.githubusercontent.com/67423319/267961939-e2fb42d5-019b-48a9-8b28-4eeda4a8978e.png)
> > > Could this be a problem with such terrible image quality?
> 
> yes, 512x512 is unusable for sdxl
how can this be fixed?  resize images in the dataset, or add an argument to the command?
